lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -1031,12 +1037,14 @@ void PatchEntryPointMutate::addUserDataArgs(SmallVectorImpl<UserDataArg> &userDa
         break;
 
       unsigned userDataValue = node.offsetInDwords;
-      if (m_pipelineState->getShaderOptions(m_shaderStage).updateDescInElf &&
-          (m_shaderStage == ShaderStageFragment || m_shaderStage == ShaderStageVertex ||
-           m_shaderStage == ShaderStageCompute)) {
+      if (m_pipelineState->getShaderOptions(m_shaderStage).updateDescInElf && m_shaderStage == ShaderStageFragment) {
         // Put set number to register first, will update offset after merge ELFs
-        // For partial pipeline compile, only fragment shader needs to adjust offset of root descriptor
-        // If there are more individual shader compile in future, we can add more stages here
+        // For partial pipeline compile, only fragment shader needs to adjust offset of root descriptor.
+        // This is part of the original "partial pipeline compile" scheme, and it uses a magic number for the
```

Comment:
Is this claim about "only the fragment shader" actually true? If we don't have the pipeline layout for a partial, relocatable compile, then the same would apply to vertex/geometry shaders, no?

Reply:
Maybe, but that is part of Qun and DavidZ's "partial pipeline compile" scheme, which I'm trying not to touch here. In the condition above, I removed the checks for vertex and compute shader to restore that bit of functionality to how it was left by DavidZ. In our new "shader/half-pipeline compile", we don't use this any more.